ext/rotoscope/rotoscope.c

```diff
@@ -119,44 +119,35 @@ static rs_class_desc_t class2str(VALUE klass)
   return real_class;
 }
 
-static char *caller_location(int start)
+static rs_callsite_t caller_location(int start)
 {
   VALUE argv[2] = {INT2NUM(start), INT2NUM(1)};
-  VALUE res = rb_funcallv(rb_mKernel, id_caller_locations, 2, argv);
-  return RSTRING_PTR(rb_inspect(rb_ary_entry(res, 0)));
+  VALUE locations = rb_funcallv(rb_mKernel, id_caller_locations, 2, argv);
+  VALUE location = rb_ary_entry(locations, 0);
+  VALUE path = rb_funcall(location, id_path, 0);
+
+  return (rs_callsite_t) {
+    .filepath = path != Qnil ? RSTRING_PTR(path) : "",
```

Comment:
Sure, I'll use `!NIL_P(path)` since that is equivalent to `path != Qnil`.  RTEST is to check if the value is truthy, if you expect that it might be `false`, which didn't seem necessary here.